patroni/ha.py

```diff
@@ -725,8 +761,23 @@ def process_unhealthy_cluster(self):
                         logger.info('Cleaning up failover key after acquiring leader lock...')
                         self.dcs.manual_failover('', '')
                 self.load_cluster_from_dcs()
-                return self.enforce_master_role('acquired session lock as a leader',
-                                                'promoted self to leader by acquiring session lock')
+
+                if self.cluster.is_standby_cluster():
+                    # standby leader disappeared, and this is a healthiest
+                    # replica, so it should become a new standby leader.
+                    # This imply that we need to start following a remote master
+                    msg = 'follow remote master {0}'.format(follow_target.conn_url)
+                    clone_target = self.cluster.get_target_to_follow()
+                    follow_target = self.cluster.get_target_to_follow()
+
+                    self._async_executor.schedule('follow_remote_master')
+                    self._async_executor.run_async(self.follow, args=(follow_target, msg))
+                    return 'promoted self to a standby leader because i had the session lock'
```

Comment:
`promoted self to a standby leader by acquiring session lock`

Reply:
This branch is within `if self.acquire_lock()`, isn't it enough?